From 9ba83271ab24ca687782880fba128d880470be6e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thomas=20M=C3=BCller?= <git@tom94.net>
Date: Tue, 3 Feb 2026 19:58:11 +0100
Subject: [PATCH] fix(compilation): flatpak compilation with AOM dependency

---
 dependencies/CMakeLists.txt | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/dependencies/CMakeLists.txt b/dependencies/CMakeLists.txt
index 7ea910b6..68898af4 100644
--- a/dependencies/CMakeLists.txt
+++ b/dependencies/CMakeLists.txt
@@ -155,14 +155,14 @@ if (TEV_SUPPORT_AVIF)
     set(AOM_PREFIX_DIR ${CMAKE_CURRENT_BINARY_DIR}/${AOM_TARGET_NAME})
     set(AOM_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/aom)
     set(AOM_BINARY_DIR ${AOM_PREFIX_DIR}/src/${AOM_TARGET_NAME}-build)
-    set(AOM_INCLUDE_DIR ${AOM_SOURCE_DIR})
+    set(AOM_INCLUDE_DIRS ${AOM_SOURCE_DIR})
     set(AOM_LIB ${CMAKE_STATIC_LIBRARY_PREFIX}aom${CMAKE_STATIC_LIBRARY_SUFFIX})
     if(IS_MULTI)
         set(AOM_LIB_PREFIX ${AOM_BINARY_DIR}/$<CONFIG>/)
     else()
         set(AOM_LIB_PREFIX ${AOM_BINARY_DIR}/)
     endif()
-    set(AOM_LIBRARY ${AOM_LIB_PREFIX}${AOM_LIB})
+    set(AOM_LIBRARIES ${AOM_LIB_PREFIX}${AOM_LIB})
 
     find_program(NASM_EXECUTABLE nasm)
     if (NASM_EXECUTABLE)
@@ -194,9 +194,26 @@ if (TEV_SUPPORT_AVIF)
             -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}
             -DCMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES}
             -DCMAKE_MSVC_RUNTIME_LIBRARY=${CMAKE_MSVC_RUNTIME_LIBRARY}
-        BUILD_BYPRODUCTS ${AOM_LIBRARY}
+        BUILD_BYPRODUCTS ${AOM_LIBRARIES}
         INSTALL_COMMAND ""
     )
+
+    set(AOM_LIBRARY ${AOM_LIBRARIES} CACHE STRING " " FORCE)
+    set(AOM_INCLUDE_DIR ${AOM_INCLUDE_DIRS} CACHE PATH " " FORCE)
+    set(AOM_DIR ${AOM_SOURCE_DIR} CACHE PATH " " FORCE)
+
+    add_library(aom-lib INTERFACE IMPORTED GLOBAL)
+    add_dependencies(aom-lib ${AOM_TARGET_NAME})
+    set_target_properties(aom-lib PROPERTIES
+        INTERFACE_INCLUDE_DIRECTORIES "${AOM_INCLUDE_DIRS}"
+        INTERFACE_LINK_LIBRARIES "${AOM_LIBRARIES}"
+    )
+
+    add_library(AOM::aom ALIAS aom-lib)
+    add_library(AOM::aom_static ALIAS aom-lib)
+    add_library(AOM::aomenc ALIAS aom-lib)
+    add_library(AOM::aomdec ALIAS aom-lib)
+    set(AOM_DECODER_FOUND TRUE CACHE BOOL " " FORCE)
 endif()
 
 # Compile libde265 (dependency of libheif, which follows)
